### PR TITLE
Add async key value unit tests

### PR DESCRIPTION
**Pull request by mstemm**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-udig

> /area libscap

/area libsinsp

/area tests

> /area proposals

**What this PR does / why we need it**:
Add unit tests for async_key_value_source, along with some potential race conditions on startup found by helgrind.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by FedeDP — DISMISSED** [review]
/approve

File: userspace/libsinsp/test/async_key_value_source.ut.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+		async_key_value_source<std::string, uint64_t>(wait_response_ms, UINT64_MAX),
+		m_delay_sec(delay_sec)
+	{}
+
+	virtual ~test_key_value_source()
+	{
+		stop();
+	}
+
+	void stopp()
```
Comment (Andreagit97) [inline userspace/libsinsp/test/async_key_value_source.ut.cpp:35]: Is this really necessary?

**Review by Andreagit97 — COMMENTED** [review]
LGTM! just a minor comment, thank you!

File: userspace/libsinsp/test/async_key_value_source.ut.cpp (hunk 1 above)
Comment (mstemm) [reply userspace/libsinsp/test/async_key_value_source.ut.cpp:35]: Oh no, it's not. It's leftover from when I was doing addl testing. Let me remove it.

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**